e2e/metrics/bundledeployment_test.go

```diff
@@ -0,0 +1,168 @@
+package metrics_test
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/rancher/fleet/e2e/metrics"
+	"github.com/rancher/fleet/e2e/testenv"
+	"github.com/rancher/fleet/e2e/testenv/kubectl"
+)
+
+var _ = Describe("BundleDeployment Metrics", Label("bundledeployment"), func() {
+	const (
+		branch    = "master"
+		namespace = "fleet-local" // required for this test to create BundleDeployments
+	)
+
+	var (
+		// kw is the kubectl command for namespace the workload is deployed to
+		kw kubectl.Command
+		// objName is going to be "randomized" instead of using a dedicated and
+		// random namespace, like it is the case for the other tests.
+		objName string
+	)
+
+	BeforeEach(func() {
+		k = env.Kubectl.Namespace(env.Namespace)
+		kw = k.Namespace(namespace)
+		objName = testenv.AddRandomSuffix(
+			"metrics",
+			rand.New(rand.NewSource(time.Now().UnixNano())),
+		)
+
+		err := testenv.CreateGitRepo(
+			kw,
+			namespace,
+			objName,
+			branch,
+			"simple-manifest",
+		)
+		Expect(err).ToNot(HaveOccurred())
+
+		DeferCleanup(func() {
+			out, err := k.Delete("gitrepo", objName)
+			Expect(err).ToNot(HaveOccurred(), out)
+		})
+	})
+
+	When("testing BundleDeployment metrics", func() {
+		bundleDeploymentMetricNames := []string{
+			"fleet_bundledeployment_state",
+		}
+		bundleDeploymentMetricStates := []string{
+			"ErrApplied",
+			"Modified",
+			"NotReady",
+			"OutOfSync",
+			"Pending",
+			"Ready",
+			"WaitApplied",
+		}
+
+		It("should have exactly one metric for the BundleDeployment", func() {
+			et := metrics.NewExporterTest(metricsURL)
+			Eventually(func() error {
+				for _, metricName := range bundleDeploymentMetricNames {
+					for _, state := range bundleDeploymentMetricStates {
+						_, err := et.FindOneMetric(
+							metricName,
+							map[string]string{
+								"name":              objName + "-simple-manifest",
+								"cluster_namespace": namespace,
+								"state":             state,
+							},
+						)
+						if err != nil {
+							return err
+						}
+					}
+				}
+				return nil
+			}).ShouldNot(HaveOccurred())
+		})
+
+		Context(
+			"when the GitRepo (and therefore Bundle) is changed",
+			Label("bundle-altered"),
+			func() {
```

Comment:
nit: newlines look strange to me
```suggestion
		When("the GitRepo (and therefore Bundle) is changed", Label("bundle-altered"), func() {
```